objects/objects.go

```diff
@@ -35,6 +35,31 @@ type Object struct {
 	Flags           uint32           `msgpack:"flags" json:"flags"`
 }
 
+// Return empty lists for nil slices.
+func (o Object) MarshalJSON() ([]byte, error) {
+	// Create an alias to avoid recursive MarshalJSON calls
+	type Alias Object
+
+	ret := Alias(o)
```

Comment:
I'd make this a pointer receiver to avoid having to copy the whole struct when serializing it; i.e.

```go
func (o *Object) MarshalJSON() ([]byte, error) {
        type Alias Object
        ret := (*Alias)(o)
        // ...
```